tracks/elixir/exercises/roman-numerals/mentoring.md

```diff
@@ -82,3 +82,51 @@ def numerals(number, acc \\ "") do
   # ...
 end
 ```
+
+#### Map keys are not ordered
+
+Some students create a map of integers to roman numerals, and then take the keys, assuming they are ordered. Such an approach would look like this:
+
+```elixir
+defmodule RomanNumerals do
+  @integer_to_roman %{
+    1 => "I",
+    4 => "IV",
+    5 => "V",
+    9 => "IX",
+    10 => "X",
+    40 => "XL",
+    50 => "L",
+    90 => "XC",
+    100 => "C",
+    400 => "CD",
+    500 => "D",
+    900 => "CM",
+    1000 => "M"
+  }
+ 
+ # ...
+
+  def numeral(number, roman_temp) do
+    Map.keys(@integer_to_roman)
+    |> Enum.reverse()
+    |> Enum.find(fn key -> number - key >= 0 end)
+    |> # more code ...
+  end
+end
+```
+
+While map keys most of the time seems to have a stable order, in reality they don't. The student could be pointed to the [documentation of the Map module](https://hexdocs.pm/elixir/Map.html) that state:
+
+> ```elixir
+> %{"one" => :two, 3 => "four"}
+> %{3 => "four", "one" => :two}
+> ```
+>
+> Key-value pairs in a map do not follow any order (that's why the printed map in the example above has a different order than the map that was created).
+
+To get rid of the problem, the student should sort the keys explicitly:
```

Comment:
To get rid of the problem, students should use an ordered container from the beginning, a list of tuples suites very well.

Reply:
Yeah, good point! I edited the notes to mention both using a list and sorting keys as a possible fix. I think it depends on context with fix should be proposed. If the code is otherwise fine, suggesting to completely change the data structure will work, but if the student is visibly struggling with other things as well, and already relies on accessing the map via `@map[key]`, explicitly sorting the keys is a simpler fix for them to understand. 

Or areee there other harms of using a map for this exercise that I am missing?

Reply:
A map is an unordered container of keys and values. And using `Access` to get values out of it already shows that one missed better opportunities to solve the exercise by a `Enum.reduce` over the list of tuples or recursion over the number and the list or some metaprogramming over the list.

Though you are right, if a student is struggling with other things as well and there is a lot of incremental rewriting necessary, then its an acceptable intermediary step, and could even be seen in an accepted solution.

Reply:
Oh, and very import, sorting the keys is only acceptable, if its done only once!

Reply:
Indeed! I added that to the notes as well.